### PR TITLE
feat: add new action option component

### PR DESCRIPTION
**Pull request by fergusjordantw**
## 🖼 Context

image

We're adding a new tab to the account page which needs us to have an option component which renders a link button to trigger the action. 

## 🚀 Changes

Adds an ActionOption which just composes the Option with a tertiary button.

## Considerations

Have removed the underline with CSS to match what's in the mockups here: https://www.figma.com/file/dDjJFIep9tYV64PtxFcJ2G/Action-Option - however I'm perfectly happy to remove this if we see fit

## ✅ Checklist

- [x] Make PR title meaningful and follow [the commit lint format](https://github.com/transferwise/neptune-web/blob/master/CONTRIBUTING.md#versioning-and-commit-lint) (it will appear in the changelog)
- [x] Changes are tested and all tests pass
- [x] Changes meet [accessibility standards](https://github.com/transferwise/neptune-web/blob/main/ACCESSIBILITY.md) and there are no violations in the console
- [x] Changes work in all supported browsers (don't forget IE11)
- [ ] You've updated the documentation if necessary

### CODE REVIEW

**Comment by mikknarusk** [comment]
maybe just me but isn't the name ActionOption a bit too generic? Anything could be an _action_, like checking a CheckboxOption or clicking on a NavigationOption. Since we're using a Button as the action, should it be ButtonOption?
cc @dozortsev @gusso

**Review by mikknarusk — APPROVED** [review]
LGTM

File: packages/components/src/actionOption/ActionOption.spec.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    const getAction = () =>
+      screen.getByRole('button', {
+        name: 'action',
+      });
+
+    const mockOnClick = jest.fn();
+
+    render(<ActionOption title="title" content="content" action="action" onClick={mockOnClick} />);
+
+    expect(getAction()).toBeInTheDocument();
```
Comment (matebek-tw) [inline packages/components/src/actionOption/ActionOption.spec.js:39]: The `screen.getBy*` selector throws an error when it cannot find the element in the DOM, so this test could be simplified:

```javascript
  expect(mockOnClick).not.toHaveBeenCalled();
  screen.getByRole('button', {
    name: 'action',
  }).click();
  expect(mockOnClick).toHaveBeenCalledTimes(1);
```

File: packages/components/src/actionOption/ActionOption.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+type Props = {
+  complex?: boolean;
+  content?: ReactChild;
+  disabled?: boolean;
+  media?: ReactChild;
+  onClick: () => void;
+  title: ReactChild;
+  action: ReactChild;
+};
```
[the remark this answers is not in the document]
Comment (fergusjordantw) [reply packages/components/src/actionOption/ActionOption.tsx:17]: Should it be passed down to option, or applied to the parent next to the "action-option" className?

File: packages/components/src/actionOption/ActionOption.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  complex,
+}: Props): ReactElement => {
+  const sharedProps = { media, title, content, complex, disabled };
+
+  return (
+    <Option
+      className="action-option"
+      {...sharedProps}
+      button={
+        <Button
```
[the remark this answers is not in the document]
Comment (fergusjordantw) [reply packages/components/src/actionOption/ActionOption.tsx:35]: resolved offline, will set to small

File: packages/components/src/actionOption/ActionOption.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (dozortsev) [reply packages/components/src/actionOption/ActionOption.tsx:17]: I would do this same way as in [`NavigationOption`](https://github.com/transferwise/neptune-web/blob/SEND-2646-add-action-option/packages/components/src/navigationOption/NavigationOption.js#L37)

**Pull request merged**